src/unit/test_util.c

```diff
@@ -283,6 +285,24 @@ static int cache_exist(int fd) {
 }
 #endif
 
+/**
+ * Get a temporary file path that is NOT located on a tmpfs (RAM-backed filesystem).
+ * This is required because we are testing fadvise(FADV_DONTNEED), which operates on disk-backed files.
+ */
+static const char *get_non_tmpfs_temp_file_path(void) {
+    struct statfs buf;
+
+    /* Check if /tmp is memory-backed (e.g., tmpfs) */
+    if (statfs("/tmp", &buf) == 0) {
+        if (buf.f_type != TMPFS_MAGIC) { // Not tmpfs, use /tmp
+            return "/tmp/redis-reclaim-cache-test";
+        }
+    }
```

Comment:
Can we just make this check part of the test_reclaimFilePageCache and return 0 in case it is not file backed? I think forcing this test to retun 0? Honestly I would also be fine with eliminating this test.